### ISSUE TITLE
Add a `name` subfield

### ISSUE THREAD

**Issue by lindseydiloreto** [doublesecretagency/craft-googlemaps#26]
Some locations have a proper name which goes beyond their street address (ie: "Empire State Building"). It may make sense to store this name alongside the existing Address data. The Google API results include an `address_components` array, which often contains a `premise`. When available, we could use this value as the associated place name.

Once this has been added, you would be able to access the location's name like this...

```twig
{{ address.name }}
```

The value should also be available as an Address subfield.

 - It should precede all other subfields by default.
 - It should be _hidden_ by default, as most addresses are not named.

**Comment by lindseydiloreto** [comment]
This FR ties very closely to #25. If they happen, they are likely to happen together.

**Comment by kerns** [comment]
I would really love this feature. It's the only important feature missing and I think a lot of projects could benefit from it. 

Small side note: I am not always in love with the official place/premise names and they are not always clean or usable. For example some businesses add them in ALL CAPS, or with an Inc / LIMITED or some other legalistic syntax that doesn't make a good or coherent label relative to how the establishment is known or referred to popularly.

For this reason, the ideal setup would be if you could type into an address name field, select from auto suggestions with enter, but then edit/sanitize the address name yourself prior to submission ...without changing the rest of the address info or otherwise breaking the universe.

If there is anything I can do in terms of testing or sparring on the feature please let me know.

**Comment by lindseydiloreto** [comment]
The [`name` subfield](https://plugins.doublesecretagency.com/google-maps/models/address-model/#name) has been added, and will be formally released with **v4.1.0**. 👍

In the meantime, you can get it by switching to the `dev` branch...

```
"doublesecretagency/craft-googlemaps": "dev-v4-dev"
```
